cmd/config-current.go

```diff
@@ -480,6 +480,12 @@ func lookupConfigs(s config.Config, setDriveCounts []int) {
 	if err != nil {
 		logger.LogIf(ctx, fmt.Errorf("Unable to setup KMS with current KMS config: %w", err))
 	}
+	if kmsCfg.Vault.Enabled {
+		const deprecationWarning = `Native Hashicorp Vault support is deprecated and will be removed on 2021-10-01. Please migrate to KES + Hashicorp Vault: https://github.com/minio/kes/wiki/Hashicorp-Vault-Keystore
+Note that native Hashicorp Vault and KES + Hashicorp Vault are not compatible.
+If you need help to migrate smoothly visit: https://min.io/pricing`
+		logger.LogIf(ctx, fmt.Errorf(deprecationWarning))
+	}
 
```

Comment:
BTW can you also remove the deprecation notice for KMS_AUTO_ENCRYPTION? @aead as we might always have it enabled for customers since we now fully support compression + encryption

Reply:
Yes, will remove the deprecation warning for auto-encryption. We will also enable encryption by default in the future. So any deployment will behave like `MINIO_KMS_AUTO_ENCRYPTION=on` - but you can opt-out via `MINIO_KMS_AUTO_ENCRYPTION=off`